Exception/ValidationFailedError.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Text;
+
+namespace Kaitai
+{
+    /// <summary>
+    ///     Common ancestor for all validation failures. Stores pointer to
+    ///     KaitaiStream IO object which was involved in an error.
+    /// </summary>
+    public class ValidationFailedError : KaitaiStructError
+    {
+        protected KaitaiStream io;
+
+        public ValidationFailedError(string msg, KaitaiStream io, string srcPath)
+            : base("at pos " + io.Pos + ": validation failed: " + msg, srcPath)
```

Comment:
IMHO: positions in the stream must be stored as a number field.

Reply:
Let's improve this in follow-ups.